src/script/interpreter.cpp

```diff
@@ -1318,6 +1318,8 @@ bool GenericTransactionSignatureChecker<T>::CheckSig(const std::vector<unsigned
     if (vchSig.empty())
         return false;
     int nHashType = vchSig.back();
+    if (((nHashType & 0x1f) == SIGHASH_SINGLE) && (nIn >= txTo->vout.size()) && require_secure_single)
```

Comment:
nit: I would lead with `require_secure_single` since it's a feature flipper

Reply:
It is ranked by rarity. The use if SIGHASH_SINGLE is rare. If it is not used, the remaining conditions will not be inspected.